app/src/main/java/me/hufman/androidautoidrive/carapp/notifications/views/DetailsView.kt

```diff
@@ -229,10 +231,28 @@ class DetailsView(val state: RHMIState, val phoneAppResources: PhoneAppResources
 			clearButton.setEnabled(false)
 		}
 
+		val hasReadoutAction = !notificationSettings.shouldReadoutNotificationDetails()
+		val readoutButton = buttons[1]
+		if (hasReadoutAction) {
+			readoutButton.apply {
+				setEnabled(true)
+				getTooltipModel()?.asRaDataModel()?.value = L.NOTIFICATION_READOUT_ACTION
+				getAction()?.asRAAction()?.rhmiActionCallback = RHMIActionButtonCallback {
+					getAction()?.asHMIAction()?.getTargetModel()?.asRaIntModel()?.value = state.id
+					readoutInteractions.triggerReadout(notification)
+				}
+			}
+		} else {
+			readoutButton.apply {
+				setEnabled(false)
+				getAction()?.asHMIAction()?.getTargetModel()?.asRaIntModel()?.value = listView.state.id
+			}
+		}
 		// enable any custom actions
-		(0..4).forEach {i ->
+		val offset = if (hasReadoutAction) 2 else 1
+		(if (hasReadoutAction) (0..3) else (0..4)).forEach {i ->
```

Comment:
I think the count index can be pulled out a bit similar to the offset:
```kotlin
val remaining = if (hasReadoutAction) 3 else 4
val offset = if (hasReadoutAction) 2 else 1
(0..remaining).forEach {
```

I'd like to see this dual-counting loop simplified but I don't know the best way. Could something be done with `forEachIndexed` perhaps?
```kotlin
val offset = if (hasReadoutAction) 2 else 1
buttons.subList(offset, buttons.size).forEachIndexed { i, button ->
	val action = notification.actions.getOrNull(i)
```

Reply:
refactored to configure all buttons that are not explicitly assigned before.